### PR TITLE
switch to slog

### PR DESCRIPTION
**Pull request by junyitang2-cyber**
slog added
slog scope added
use info to replace println in check.rs

### CODE REVIEW

File: src/tools/kata-ctl/src/main.rs [hunk 1]
```diff
@@ -20,7 +21,24 @@ use ops::check_ops::{
...
+    () => {
+        slog_scope::logger()
+    };
+}
+
 fn real_main() -> Result<()> {
+    //create logger
+    let log_level_name = global_args
+        .value_of("log-level")
+        .ok_or_else(|| anyhow!("cannot get log level"))?;
```
Comment (jodh-intel) [inline src/tools/kata-ctl/src/main.rs:35]: For the log level, you'll either have to hard-code it for this PR as something like `let log_level_name = "info"`. Or, better, you could add a `--log-level <level>` command line option. However, to do that will take a bit of reading:

- https://docs.rs/clap/latest/clap/_derive/_tutorial
- https://docs.rs/clap/latest/clap/_derive/index.html

In summary, you'll need to modify `KataCtlCli` in `src/args.rs` to add a `pub LogLevel: String` field as we would want to be able `kata-ctl` like this:

```bash
$ kata-ctl $command ...                     # We currently have this.
$ kata-ctl --log-level $level $command ...  # This is what we'd ideally like to be able to do.
```

File: src/tools/kata-ctl/src/main.rs [hunk 2]
```diff
@@ -20,7 +21,24 @@ use ops::check_ops::{
...
+
+    let log_level = logging::level_name_to_slog_level(log_level_name).map_err(|e| anyhow!(e))?;
+
+    let writer = io::stdout();
+    let (logger, _guard) = logging::create_logger(name, crate_name!(), log_level, writer);
```
Comment (jodh-intel) [inline src/tools/kata-ctl/src/main.rs:40]: You'll need to define `name`, or just do something like:

```suggestion
    let (logger, _guard) = logging::create_logger(crate_name!(), crate_name!(), log_level, writer);
```

**Review by jodh-intel — COMMENTED** [review]
Thanks @junyitang2-cyber - a few comments...

Also, please can you ensure you can run `make` successfully in the `src/tools/kata-ctl` directory. That builds the tool in the standard way so even if `cargo build` works, `make` may not as we enable extra checks.

**Comment by jodh-intel** [comment]
Hi @junyitang2-cyber - please could you rebase this PR to resolve the conflicts?

File: src/.idea/.gitignore [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# 默认忽略的文件
```
[the remark this answers is not in the document]
Comment (junyitang2-cyber) [reply src/.idea/.gitignore:1]: sure, working on it

**Comment by jodh-intel** [comment]
Hi @junyitang2-cyber - I assume you are no longer working on this PR, so I am going to resolve the conflicts and create a new PR based on this one using our [assisted PR process](https://github.com/kata-containers/community/wiki/Assisted-PR-process).

Please do let me know asap if you do still plan to work on fixing the issues on this PR though! :smile:

**Comment by jodh-intel** [comment]
Related: #5401.

I'm assigning this to the [current outreach team](https://github.com/kata-containers/community/wiki/Outreach---Internship-programs).

**Pull request closed without merging**